valuetypes.go

```diff
@@ -514,3 +516,123 @@ func (d DateTime) Map(data interface{}, target interface{}) error {
 		return ErrTypeMismatch
 	}
 }
+
+// Duration schema type for duration as integer seconds or string on the form:
+//
+//     /[-+]? (\d+ d(ays?)?)? (\d+ h((ours?)?|r))? (\d+ m(in(untes?)?)?)?/
+//
+// This allows for a lot of human readable time durations, examples:
+//
+//    '31 days 2 hours 5 min'
+//    '5 min'
+//    '2 hours 5 min'
+//    '+ 3 minutes'
+//    '- 2 hr'
+//    '- 3 days'
+//    '1d2h3m'
+//    '   1  day  2 hour  1 minutes '
+type Duration struct {
+	MetaData
+	AllowNegative bool
+}
+
+var durationPattern = strings.Join([]string{
+	`(?:\s*(\d+)\s*d(?:ays?)?)?`,
+	`(?:\s*(\d+)\s*h(?:ours?|r)?)?`,
+	`(?:\s*(\d+)\s*m(?:in(?:utes?)?)?)?`,
+}, "")
+var durationRegexp = regexp.MustCompile(`^\s*` + durationPattern + `\s*$`)
+var signedDurationRegexp = regexp.MustCompile(`^\s*([+-])?` + durationPattern + `\s*$`)
+
+// Schema returns a JSON representation of the schema.
+func (d Duration) Schema() map[string]interface{} {
+	m := d.schema()
+	m["type"] = []string{"integer", "string"}
+	if d.AllowNegative {
+		m["pattern"] = signedDurationRegexp.String()
+	} else {
+		m["pattern"] = durationRegexp.String()
+	}
+	return m
+}
+
+// Validate the given data, this will return nil if data satisfies this schema.
+// Otherwise, Validate(data) returns a ValidationError instance.
+func (d Duration) Validate(data interface{}) error {
```

Comment:
I'm not a big fan of taking an arbitrary type for an input. I know it works, but it isn't idiomatic, it is more like python than go. Is this because Duration is satisfying some interface which requires it?